diffsims/sims/kinematic_simulation.py

```diff
@@ -0,0 +1,187 @@
+"""
+Created on 1 Nov 2019
+
+Back end for computing diffraction patterns with a kinematic model.
+
+@author: Rob Tovey
+"""
+from diffsims.utils.discretise_utils import getDiscretisation
+from numpy import array, pi, sin, cos, empty, maximum, sqrt
+from scipy.interpolate import interpn
+from diffsims.utils.fourier_transform import (getDFT, toRecip, fftshift_phase,
+    plan_fft, fast_abs)
+from diffsims.utils.generic_utils import toMesh
+
+
+def normalise(arr): return arr / arr.max()
+
+
+def get_diffraction_image(coordinates, species, probe, x, wavelength,
+                          precession, GPU=True, pointwise=False, **kwargs):
+    """
+    Return kinematically simulated diffraction pattern
+
+    Parameters
+    ----------
+    coordinates : ndarray of shape [n_atoms, 3] and floating point type
+        List of atomic coordinates
+    species : ndarray of shape [n_atoms] and integer type
+        List of atomic numbers
+    probe : instance of probeFunction
+        Function representing 3D shape of beam
+    x : list of length 3 of 1D ndarrays
+        Mesh on which to compute the volume density
+    wavelength : float
+        Wavelength of electron beam
+    precession : a pair (float, int)
+        The float dictates the angle of precession and the int how many points are
+        used to discretise the integration.
+    dtype : (str, str)
+        tuple of floating/complex datatypes to cast outputs to
+    ZERO : float > 0
+        Rounding error permitted in computation of atomic density. This value is
+        the smallest value rounded to 0.
+    GPU : bool
+        Flag whether to use GPU or CPU discretisation. Default (if available) is True
+    pointwise : bool
+        Optional parameter whether atomic intensities are computed point-wise at
+        the centre of a voxel or an integral over the voxel. default=False
+
+    Returns
+    -------
+    DP : ndarray
+        The two-dimensional diffraction pattern evaluated on the reciprocal grid
+        corresponding to the first two vectors of <x>.
+    """
```

Comment:
I think (may be misremembering) that this doesn't comply with the numpy docstring standards - isn't it after the : it should just be the type and details regarding shape etc should all be in the next line?

Reply:
Sorry, I wasn't following any formal standards, I'll go back and do that.